src/views/pages/product/single.twig

```diff
@@ -80,6 +80,7 @@
 | product.add_to_cart_label              | string                      |                                                                        |
 | product.currency                       | string                      |                                                                        |
 #}
+{% set product_is_on_sale = product.is_on_sale and product.regular_price > product.sale_price %}
```

Comment:
no need for this, use it as is, `product.is_on_sale` if there is edge case related to it, should be done from back end